### PR TITLE
blueprint: Configure Spark's memory

### PR DESCRIPTION
**Pull request by kayousterhout**
By default, Spark will give each executor JVM 1GB of memory. This
commit explicitly configures the executor memory, so that users can
pass in the amount of memory they'd like to allocate to each Spark
executor, and changes sparkRun.js to automatically set Spark's
memory based on the machines being used.

This PR relies on https://github.com/kelda/kelda/pull/1260

### CODE REVIEW

**Comment by kayousterhout** [comment]
@ejj I'm curious what you think of this approach.  An alternative strategy is that we could pass the set of machines that Spark will run on as a parameter to the Spark cluster, and Spark could use those to automatically determine the correct number of workers and memory.  I didn't do that because it seems slightly misleading, because we won't actually schedule Spark on those workers (so a user could pass in a subset of workers in her cluster, expecting Spark to actually run on those workers, and in reality the containers would be balanced across the cluster)

**Comment by ejj** [comment]
I don't think I understand the problem well enough to have a preference either way.  IMO this seems like a reasonable first approach.

File: spark.js [hunk 1]
```diff
@@ -14,20 +14,28 @@ function setImage(newImage) {
 }
 
 /**
+ * @param {number} memoryGb - The amount of memory that each Spark executor should
```
Comment (luise) [inline spark.js]: Should this maybe be memory GB since Gb has a different meaning?
And in the description, maybe say "The amount of memory in GB that..."

Comment (kayousterhout) [reply spark.js]: Ah good point fixed!

**Pull request merged**